crates/stages/src/stage.rs

```diff
@@ -58,6 +69,37 @@ impl ExecInput {
         let is_final_range = end == target;
         (start..=end, is_final_range)
     }
+
+    /// Return the next block range determined the number of transactions within it.
+    /// This function walks the the block indices until either the end of the range is reached or
+    /// the number of transactions exceeds the threshold.
+    pub fn next_block_range_with_transaction_threshold<DB: Database>(
+        &self,
+        tx: &Transaction<'_, DB>,
+        tx_threshold: u64,
+    ) -> Result<(RangeInclusive<TxNumber>, BlockNumber, bool), StageError> {
```

Comment:
maybe return the block range, as we do in other functions for consistency?
```suggestion
    ) -> Result<(RangeInclusive<TxNumber>, RangeInclusive<BlockNumber>, bool), StageError> {
```

or change the function comment to reflect that we return only the end of the range